app/controllers/application_controller.rb

```diff
@@ -77,11 +77,17 @@ def ensure_visit_integrity
     end
   end
 
-  def pad_execution_time(execution_time)
+  def pad_execution_time(execution_time_limit, max_pad_multiple=5)
     start = Time.zone.now
     result = yield
     stop = Time.zone.now
-    sleep execution_time - (stop - start)
+    pad_time = execution_time_limit - (stop - start)
+    if pad_time < 0
+      pad_time = rand(max_pad_multiple) * execution_time_limit
+      Rails.logger.info "Execution padding insufficient"
```

Comment:
It comes out like this:

```json
{
  "_index": "logstash-2015.10.13",
  "_type": "pvb",
  "_id": "2d0f34af-1a3d-3aed-6a72-b6e5e04960b9",
  "_score": 4.2507687,
  "_source": {
    "tags": [
      "ruby",
      "application",
      "production",
      "pvb"
    ],
    "@version": 1,
    "@timestamp": "2015-10-13T21:37:12.327+00:00",
    "host": "front-01.prod1.pvb",
    "file": "/srv/pvb/application/shared/log/production.log",
    "message": "I, [2015-10-13T21:37:11.224153 #13003]  INFO -- : Deserializing legacy token",
    "type": "pvb",
    "@logstash_uuid": "2d0f34af-1a3d-3aed-6a72-b6e5e04960b9"
  }
}
```

In the case of the legacy tokens, I just used the literal text to select entries for the graph.